pitop/robotics/navigation/navigation_controller.py

```diff
@@ -0,0 +1,323 @@
+from threading import Thread, Event
+import time
+from dataclasses import dataclass
+import numpy as np
+import math
+from typing import Union
+from simple_pid import PID
+import sched
+
+
+@dataclass
+class RobotState:
+    x: float = 0.0
+    y: float = 0.0
+    theta: float = 0.0
+    v: float = 0.0
+    w: float = 0.0
+
+    @property
+    def pose(self):
+        return self.x, self.y, self.theta
```

Comment:
Not sure about returning multiple values like this. What is a 'pose', anyway?

Reply:
pose is every variable required to estimate a robot's position and orientation in space. In our case, we're on a 2D plane so just x, y and theta is required. I've deleted this anyway as I was getting in a mess with angles (giving user degrees and having radians internally) - I've just changed to `position` that gives x and y as this is consistent with the `go_to` arguments of `position` and `angle`